### PR TITLE
Fix compatibility issues with Rails 5.1

### PR DESCRIPTION
**Pull request by RickCSong**
In Rails 5.1, #extract_schema_qualified_name is redefined in the
PostgreSQLAdapter.  Because CockroachDBAdapter defines it slightly
differently, this causes compatibility issues between the two.  This
change ensures that the namespaces are no longer colliding.

This also restores the #index_name_length and #table_alias_length methods which
were changed in ActiveRecord 5.1 to use `SHOW max_identifier_length` which does not
exist in CockroachDB.  We hardcode this to 63 (as it was previously in
ActiveRecord 5.0).

### CODE REVIEW

**Comment by tamird** [comment]
Hi Rick! 👋

File: lib/active_record/connection_adapters/cockroachdb_adapter.rb [hunk 1]
```diff
@@ -117,7 +120,8 @@ def column_definitions(table_name)
       end_sql
   end
 
-  private def extract_schema_qualified_name(*args)
-    ActiveRecord::ConnectionAdapters::PostgreSQL::Utils.extract_schema_qualified_name(*args)
+  def table_alias_length
+    63
```
Comment (RickCSong) [inline lib/active_record/connection_adapters/cockroachdb_adapter.rb]: :+1: Sounds great.  I'll add a comment for this :)

**Comment by RickCSong** [comment]
Updated with comments @tamird !  Thanks for feedback.

File: lib/active_record/connection_adapters/cockroachdb_adapter.rb [hunk 2]
```diff
@@ -117,7 +124,17 @@ def column_definitions(table_name)
       end_sql
   end
 
-  private def extract_schema_qualified_name(*args)
-    ActiveRecord::ConnectionAdapters::PostgreSQL::Utils.extract_schema_qualified_name(*args)
+  # This is hardcoded to 63 (as previously was in ActiveRecord 5.0) to aid in
+  # migration from PostgreSQL to Cockroachdb. In practice, this limitation
+  # is arbitrary since CockroachDB supports index name lengths and table alias
+  # lengths far greater than this value.  For the time being though, we match
```
Comment (tamird) [inline lib/active_record/connection_adapters/cockroachdb_adapter.rb]: two spaces after this period

File: lib/active_record/connection_adapters/cockroachdb_adapter.rb [hunk 3]
```diff
@@ -117,7 +124,17 @@ def column_definitions(table_name)
...
+  # the original behavior for PostgreSQL to simplify migrations.
+  #
+  # Note that in the migration to ActiveRecord 5.1, this was changed in
+  # PostgreSQLAdapter to use `SHOW max_identifier_length` (which does not
+  # exist in CockroachDB).  Therefore, we have to redefine this here.
```
Comment (tamird) [inline lib/active_record/connection_adapters/cockroachdb_adapter.rb]: ditto

**Comment by jordanlewis** [comment]
LGTM once Tamir's comments are resolved. Thanks!

**Comment by RickCSong** [comment]
Done!  Thanks

**Pull request merged**